src/bap_mc/bap_mc.ml

```diff
@@ -170,6 +199,20 @@ module Cmdline = struct
     Arg.(value & opt_all ~vopt:`bil (enum formats) [] &
          info ["show-bil"] ~doc)
 
+  let addr =
+    let doc = "Specify an address of first byte, as though \
+	       the instructions occur at a certain address, \
+	       and accordingly interpreted. " in
+    Arg.(value & opt ~vopt:"0" (string) "0" &
+	 info ["addr"] ~doc)    
+
+  let max_insns =
+    let doc = "Specify a number of instructions to disassemble.\
+	       Good for ensuring that only one instruction is ever\
+	       lifted or disassembled from a byte blob. Default is all" in
+    Arg.(value & opt ~vopt:(Some 0) (some int) (Some 0) &
```

Comment:
In my case, if I omit a value to max-insns option or use --max-insns 0, there are no instructions disassembled because check evaluates to false in that case. Also, check does regard None to be a moniker for infinity, because it always returns true.

I don't dispute the argument defaults are error prone and that this is more correct, I've adopted the advice in argument handling in each case :) Thanks!